packages/api-client/src/views/Request/Request.vue

```diff
@@ -66,39 +67,24 @@ const executeRequest = async () => {
     server: activeServer.value,
   })
 
-  requestAbortController.value = controller
-
-  // TODO: Do we need this?
-  //   if (request && response) {
-  //     requestMutators.edit(activeRequest.value.uid, 'history', [
-  //       ...activeRequest.value.history,
-  //       {
-  //         request,
-  //         response,
-  //         timestamp: Date.now(),
-  //       },
-  //     ])
-  //     requestStatusBus.emit('stop')
-  //   } else {
-  //     // Toast if not cancelled by user and we have no response
-  //     if (!(error instanceof DOMException && error.name == 'AbortError'))
-  //       toast(error?.message ?? 'Send Request Failed', 'error')
-
-  //     requestStatusBus.emit('abort')
-  //   }
-  // } catch (error) {
-  //   console.error(error)
-  //   toast(`Oops! \n${error}`, 'error')
-  //   requestStatusBus.emit('abort')
-  // }
-
-  const result = await sendRequest()
-
-  requestHistory.push(result)
+  // Error from createRequestOperation
+  if (error) {
+    toast(error.message, 'error')
+    return
+  }
+
+  requestAbortController.value = requestOperation.controller
+  const [sendRequestError, result] = await requestOperation.sendRequest()
+
+  // Send error toast
+  if (sendRequestError) toast(sendRequestError.message, 'error')
+  else requestHistory.push(result)
 }
 
 /** Cancel a live request */
-const cancelRequest = async () => requestAbortController.value?.abort()
+const cancelRequest = async () =>
+  requestAbortController.value?.abort('User has cancelled the request')
```

Comment:
what do you think about that as the action is triggered by the user?

```suggestion
  requestAbortController.value?.abort('The request has been cancelled')
```

Reply:
Sounds good to me!